src/Halcyon.WebApi/HAL/Json/JsonHALMediaTypeFormatter.cs

```diff
@@ -56,8 +56,25 @@ public class JsonHALMediaTypeFormatter : JsonMediaTypeFormatter {
                     value = halResponse.ToJObject(serializer);
                 }
             }
-            
+
             return base.WriteToStreamAsync(type, value, writeStream, content, transportContext);
         }
+
+        private bool TryGetHalResponse(Type type, object value, out HALResponse response) {
+            if (_converters != null) {
```

Comment:
Should this check if the value is null? Perhaps there should be a null check at the start that just returns false?